lib/protobuf.ex

```diff
@@ -54,6 +54,15 @@ defmodule Protobuf do
         Protobuf.Builder.new!(__MODULE__, attrs)
       end
 
+      def new_and_verify!(attrs) do
+        struct = Protobuf.Builder.new!(__MODULE__, attrs)
+
+        # Will raise if unable to encode.
+        Protobuf.Encoder.encode(struct)
```

Comment:
we don't want to encode to verify, is there a different approach?

Reply:
why? performance?

Reply:
I think ideally you'd recursively walk the struct and check the types but I don't know if we're going to do that